web-console/src/views/query-view/query-view.tsx

```diff
@@ -205,14 +209,23 @@ export class QueryView extends React.PureComponent<QueryViewProps, QueryViewStat
 
         const query = QueryView.isJsonLike(queryString) ? Hjson.parse(queryString) : queryString;
 
+        const queryId = generateQueryId();
+
+        const isSql = !QueryView.isJsonLike(queryString);
+
         let context: Record<string, any> | undefined;
         if (!isEmptyContext(queryContext) || wrapQueryLimit || mandatoryQueryContext) {
           context = { ...queryContext, ...(mandatoryQueryContext || {}) };
+          context.sqlQueryId = queryId;
```

Comment:
where are you setting the `context.queryId` if this is not a SQL query?

Reply:
My mistake, i didn't setting the sqlQueryId and it still be able to cancel the query. Added it now